BoxContentSDK/BoxContentSDK/Operations/BOXAPIOperation.m

```diff
@@ -348,6 +350,16 @@ - (void)executeOperation
     }
 }
 
+- (void)setSessionTask:(NSURLSessionTask *)sessionTask
+{
+    NSURLSessionTask *oldSessionTask = _sessionTask;
+    _sessionTask = sessionTask;
+    if (_sessionTaskReplacedBlock != nil) {
+        _sessionTaskReplacedBlock(oldSessionTask, _sessionTask);
```

Comment:
we should notify whenever there is new sessionTask regardless of oldSessionTask being nil or not. We could check if oldSessionTask != _sessionTask but not sure if it's really needed.